### ISSUE TITLE
ReferenceError: XMLHttpRequest is not defined

### ISSUE THREAD

**Issue by dwilt** [andyrichardson/passport-workos#3]
Hey @andyrichardson, another question for you.. I'm using your package inside of a NestJS project. Here's how my strategy looks:

```ts
import { Injectable } from '@nestjs/common';
import { ConfigService } from '@nestjs/config';
import { PassportStrategy } from '@nestjs/passport';
// import { Options as StrategyOptions, WorkOSSSOStrategy, Verify as VerifyFunction } from './workossso.strategy';
import { WorkOSSSOStrategy } from 'passport-workos';
// import { StrategyOptions, VerifyFunction } from "passport-oauth2";

@Injectable()
export class AirWorkOSSSOStrategy extends PassportStrategy(
  WorkOSSSOStrategy,
  'workos',
) {
  constructor(configService: ConfigService) {
    super(
      {
        clientID: configService.get('WORKOS_CLIENT_ID'),
        clientSecret: configService.get('WORKOS_API_KEY'),
        callbackURL: 'http://localhost:4001/api/v1/auth/workos/callback',
      },
      // @ts-expect-error
      (_req, _accessToken, _refreshToken, profile, done) => {
        return done(undefined, profile);
      },
    );
  }
}
```
When I hit the `/callback` url, I get the following error in the server logs:

```
/Users/dwilt/Documents/GitHub/air-public-api/node_modules/passport-workos/dist/index.js:749
          return new Promise(function (t, r) {
                 ^
ReferenceError: XMLHttpRequest is not defined
    at /Users/dwilt/Documents/GitHub/air-public-api/node_modules/passport-workos/dist/index.js:754:21
    at new Promise (<anonymous>)
    at H (/Users/dwilt/Documents/GitHub/air-public-api/node_modules/passport-workos/dist/index.js:749:18)
    at Z (/Users/dwilt/Documents/GitHub/air-public-api/node_modules/passport-workos/dist/index.js:997:31)
    at ue.request (/Users/dwilt/Documents/GitHub/air-public-api/node_modules/passport-workos/dist/index.js:1303:9)
    at ue.<computed> [as post] (/Users/dwilt/Documents/GitHub/air-public-api/node_modules/passport-workos/dist/index.js:1325:19)
    at Function.post (/Users/dwilt/Documents/GitHub/air-public-api/node_modules/passport-workos/dist/index.js:443:16)
    at t.WorkOS.<anonymous> (/Users/dwilt/Documents/GitHub/air-public-api/node_modules/passport-workos/dist/index.js:2754:38)
    at Generator.next (<anonymous>)
    at /Users/dwilt/Documents/GitHub/air-public-api/node_modules/tslib/tslib.js:118:75

```
I wanted to try and debug what was going on so I copied your `src/index.ts` file directly into our project and the error went away. I'm thinking it has something to do with the way the bundled `index.js` (not `index.umd.js` or `index.modern.js` or `index.es.js`) cause why else would the native .ts work when directly used in the project.. any ideas?

**Comment by andyrichardson** [comment]
Hey @dwilt thanks for the report.

I double checked the release and it looks like the incorrect build target was being used (hence why importing the TS file is working for you). 

I've published a new beta release `0.0.2-beta.0` - haven't had a chance to test it but give that a shot and let me know if that fixes the issue.

You can install it with `npm i passport-woros@beta`

**Comment by dwilt** [comment]
@andyrichardson yep, that fixed it!

**Comment by dwilt** [comment]
@andyrichardson just so I can learn.. what about [changing the version](https://github.com/andyrichardson/passport-workos/commit/a014cb14a11c7c0f60de38da2ccd5c9ae44c52b6) fixes that?

**Comment by andyrichardson** [comment]
> @andyrichardson just so I can learn.. what about [changing the version](https://github.com/andyrichardson/passport-workos/commit/a014cb14a11c7c0f60de38da2ccd5c9ae44c52b6) fixes that?

Because it was a build change that needed fixing, there isn't really anything to see in the commit history. For the initial release, I ran something along the lines of this

```sh
$(npm bin)/microbundle build
npm publish
```

This built the package with the browser as the target runtime [(see here)](https://github.com/developit/microbundle/blob/master/README.md?plain=1#L330) which is why it was trying to use web globals such as `XMLHttpRequest`. If the target is node, it should instead use something such as the [`http` module](https://nodejs.org/api/http.html).

I'm now running

```sh
$(npm bin)/microbundle build --target node
npm publish
```

This resolves the issue! Hope that helps and thanks for the report :pray:

**Comment by dwilt** [comment]
Ahhhhhhhh, yep, got it. Thanks!!!!

**Comment by andyrichardson** [comment]
Just published [v0.0.3](https://github.com/andyrichardson/passport-workos/releases/tag/v0.0.3) which contains this fix and another bugfix.
